### ISSUE TITLE
[Feature] Patterns library

### ISSUE THREAD

**Issue by KillyMXI** [WerWolv/ImHex#407]
### What feature would you like to see?

Basically, https://github.com/kaitai-io/kaitai_struct_formats

Or at least put infrastructure and some basic formats in place, to begin with.

When there is a template language implemented in a hex editor - I also expect to see:

- good documentation (not quite there yet, some incomplete, some already outdated);
- working examples shipped with the app - to learn from and to use on common file types (I can see an empty folder so far);
- potential for knowledge accumulation (it's so much easier to contribute when the way is paved).

### How will this feature be useful to you and others?

Since you've set to invent 14+1st format for binary patterns (#61) and both collective and personal knowledge of Kaitai struct is not transferrable to this otherwise intriguing new tool, we need something to work with and learn from.

Examples improve the learning curve. If you want your app to be popular - time to think how it looks in new users eyes.

Shipping with patterns for some common file formats can also improve the usefulness for certain scenarios.

A library of common formats will likely serve as a good benchmark for the pattern language. Before I see it implementing practical examples on the same level as Kaitai - I can't be sure whether it is powerful enough for my personal task.

### Request Type

- [ ] I can provide a PoC for this feature or am willing to work on it myself and submit a PR

**Comment by WerWolv** [comment]
Hey.

The documentation about the pattern language that you can find [here](https://imhex.werwolv.net/docs) is basically complete with all features that's currently implemented. Nothing's really outdated there, if then only the current nightly features are missing.

There's a standard library as well as a bunch of patterns for common file types that can be found here: https://github.com/WerWolv/ImHex-Patterns
All files that are in this repo can also be downloaded directly from ImHex through the Content Store that can be found under `Help -> Content Store`.

Furthermore, there's a repo with basic exercises I started working on but didn't get super far yet sadly
https://github.com/WerWolv/ImHex-Patties

tldr; There's a lot of things done already. I'd just like to mention that I'm working on this by myself in my free time next to my full-time job. Without help, this simply takes time.

**Comment by KillyMXI** [comment]
> `Help -> Content Store`

Is empty for me. Reload button does nothing.

Other that that, the app's start screen, the web page and the repo Readme can benefit from some refinement. So far I can only see the Readme mentioning there is something called Content Store, with no mention of what kind of files there are. Opening it in the app and seeing it empty didn't help either - it's easy to assume it is incomplete feature.

I can't see anything linking to the patterns repo. Exploring your whole profile is a stretch. Maybe if you move ImHex to a separate organization it will be clear there is something to look for without risking to get lost in countless one-pr-forks.

Another suggestion - put buttons/links opening Content Store to places where I might be looking for something obtainable from there. Like patterns list for example. Besides Browse button there should be a Content Store button...
